### PR TITLE
Fix #3460: in 1.4.x

### LINKED ISSUE #3460: 'Unable to mount volumes for pod "syndesis-upgrade-1.4.7...' during upgrade from 1.4.6 to 1.4.7 on OSO pro
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem
I'm unable to upgrade syndesis on openshift pro:

3:31:18 PM | Warning | Failed Mount | Unable to mount volumes for pod "syndesis-upgrade-1.4.7_upgrade-project(ee1e559b-a9fb-11e8-937e-125b034d2f46)": timeout expired waiting for volumes to attach/mount for pod "upgrade-project"/"syndesis-upgrade-1.4.7". list of unattached/unmounted volumes=[backup-dir]5 times in the last 10 minutes
-- | -- | -- | --
3:20:13 PM | Normal | Successful Mount Volume | MountVolume.SetUp succeeded for volume "syndesis-operator-token-jwf4x"
3:20:13 PM | Warning | Failed Attach Volume | Multi-Attach error for volume "pvc-655fe43a-a9fb-11e8-ba69-12b5519f9b58" Volume is already used by pod(s) syndesis-db-1-2m8n5
3:20:13 PM | Normal | Scheduled | Successfully assigned syndesis-upgrade-1.4.7 to ip-172-31-58-128.ec2.internal

screen shot 2018-08-27 at 15 26 39

## Tasks involved / Steps to Reproduce

1. deploy syndesis 1.4.6 on OSO Pro using template
2. manually create operator service account 
3. try to upgrade to 1.4.7 via template ("old way")

### PR DESCRIPTION
**Pull request by nicolaferraro**
This fixes #3460 on 1.4.x.
~~I've picked up the SAR project change because it was conflicting.~~
I've also added a new commit that fixes the volume env in the pod template, that I'm going to push on master too.

Tested on Minishift.

### CODE REVIEW

**Comment by nicolaferraro** [comment]
Error is unrelated:
```
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project io.syndesis:syndesis-parent:1.4-SNAPSHOT (/workspace/app/pom.xml) has 1 error
[ERROR]     Non-resolvable import POM: Could not find artifact org.apache.camel:camel-spring-boot-dependencies:pom:2.21.0.fuse-710021 in jboss-fuse (https://origin-repository.jboss.org/nexus/content/groups/ea) @ line 966, column 19 -> [Help 2]
```

Version `2.21.0.fuse-710021` is not pinned. CI cannot work with those dependencies.

**Comment by oscerd** [comment]
That build wasn't pinned on the Jboss EA Repo, so we don't have it.

https://origin-repository.jboss.org/nexus/content/groups/ea/org/apache/camel/camel-core/

The last pinned Camel for Fuse 7.1.x is the 018 build from August 7.

**Comment by oscerd** [comment]
@cunningt we may need a new build of Fuse 7 7.1.x and we may need to pin the new Camel artifacts

File: install/generator/01-header.yml.mustache [hunk 1]
```diff
@@ -30,6 +30,10 @@ parameters:
   description: The name of the OpenShift project Syndesis is being deployed into.
   displayName: OpenShift project to deploy into
   required: true
+- name: SAR_PROJECT
```
Comment (rhuss) [inline install/generator/01-header.yml.mustache]: Not sure whether we want to have this for 1.4.x, too. This is another feature which requires also an update of the installation scripts (i.e. that template parameter is mandatory and should default to the OpenShift project). 

We should not add this to 7.1.

File: install/generator/04-syndesis-oauth-proxy.yml.mustache [hunk 2]
```diff
@@ -94,7 +94,7 @@
             - --skip-auth-preflight
             - --openshift-ca=/etc/pki/tls/certs/ca-bundle.crt
             - --openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
-            - --openshift-sar={"namespace":"${OPENSHIFT_PROJECT}","resource":"pods","verb":"get"}
+            - --openshift-sar={"namespace":"${SAR_PROJECT}","resource":"pods","verb":"get"}
```
Comment (rhuss) [inline install/generator/04-syndesis-oauth-proxy.yml.mustache]: See comment above.

File: install/operator/pkg/syndesis/configuration/configuration.go [hunk 3]
```diff
@@ -112,6 +117,9 @@ var (
 
 		metaMemoryLimitFromEnv,
 		metaVolumeCapacityFromEnv,
+		sarNamespaceFromEnv,
```
Comment (rhuss) [inline install/operator/pkg/syndesis/configuration/configuration.go]: I guess this is also relevant to the sarNamespace in the templates and creeped in somehow.

File: install/operator/pkg/syndesis/configuration/configuration.go [hunk 4]
```diff
@@ -166,11 +173,29 @@ func openShiftConsoleUrlFromEnv(config map[string]string, syndesis *v1alpha1.Syn
 
 func envOpenshiftProject(syndesis *v1alpha1.Syndesis) *SyndesisEnvVarConfig {
 	return &SyndesisEnvVarConfig{
-		Var: EnvOpenshiftProject,
+		Var:   EnvOpenshiftProject,
 		Value: syndesis.Namespace,
 	}
 }
 
+func sarNamespaceFromEnv(config map[string]string, syndesis *v1alpha1.Syndesis) {
```
Comment (rhuss) [inline install/operator/pkg/syndesis/configuration/configuration.go]: dito

File: install/syndesis-dev.yml [hunk 5]
```diff
@@ -33,6 +33,10 @@ parameters:
   description: The name of the OpenShift project Syndesis is being deployed into.
   displayName: OpenShift project to deploy into
   required: true
+- name: SAR_PROJECT
```
Comment (rhuss) [inline install/syndesis-dev.yml]: see above

File: install/syndesis-dev.yml [hunk 6]
```diff
@@ -928,7 +951,7 @@ objects:
             - --skip-auth-preflight
             - --openshift-ca=/etc/pki/tls/certs/ca-bundle.crt
             - --openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
-            - --openshift-sar={"namespace":"${OPENSHIFT_PROJECT}","resource":"pods","verb":"get"}
+            - --openshift-sar={"namespace":"${SAR_PROJECT}","resource":"pods","verb":"get"}
```
Comment (rhuss) [inline install/syndesis-dev.yml]: dito

File: install/syndesis.yml [hunk 7]
```diff
@@ -33,6 +33,10 @@ parameters:
   description: The name of the OpenShift project Syndesis is being deployed into.
   displayName: OpenShift project to deploy into
   required: true
+- name: SAR_PROJECT
```
Comment (rhuss) [inline install/syndesis.yml]: dito

File: install/syndesis.yml [hunk 8]
```diff
@@ -926,7 +949,7 @@ objects:
             - --skip-auth-preflight
             - --openshift-ca=/etc/pki/tls/certs/ca-bundle.crt
             - --openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
-            - --openshift-sar={"namespace":"${OPENSHIFT_PROJECT}","resource":"pods","verb":"get"}
+            - --openshift-sar={"namespace":"${SAR_PROJECT}","resource":"pods","verb":"get"}
```
Comment (rhuss) [inline install/syndesis.yml]: dito

**Comment by nicolaferraro** [comment]
Removed SAR part.

**Pull request merged**